client.go

```diff
@@ -25,44 +25,52 @@ var clientServerConnection net.Conn
 var clientMasterConnection rpc.Server
 //Need a map from clientId->serverConnPort
 
+// client LogicalClock
+var clientLogicalClock = 0
 
 type ClientMaster int
 
 type PutArgs struct {
 	key, value string
-	clientId, serverId int
+	clientId, serverId, clientLogicalClock int
 }
 
 type GetArgs struct {
 	key string
 }
 
+type history struct {
+	action string
+	key, value string
+	clientId, serverId int
+	timeStamp map[int]int
+}
+
+var clientHistory []history
+
 var clientWaitGroup sync.WaitGroup
 
 func (t *ClientMaster) clientPut(clientId int, key string, value string) error {
```

Comment:
the function names exported should start with a capital letter. For example, here it should be ClientPut.